lib/matplotlib/axes/_axes.py

```diff
@@ -3585,15 +3586,36 @@ def apply_mask(arrays, mask): return [array[mask] for array in arrays]
                 line = mlines.Line2D(x_masked, y_masked,
                                      marker=hlmarker, **eb_cap_style)
                 line.set(**{f"{dep_axis}data": hl_masked})
-                caplines.append(line)
+                caplines[dep_axis].append(line)
                 if capsize > 0:
-                    caplines.append(mlines.Line2D(
+                    caplines[dep_axis].append(mlines.Line2D(
                         x_masked, y_masked, marker=marker, **eb_cap_style))
-
-        for l in caplines:
-            self.add_line(l)
+        if self.name == 'polar':
+            for axis in caplines:
+                if not caplines[axis]:
+                    continue
+                lo, hi = caplines[axis]
+                for (lo_theta, lo_r,
+                     hi_theta, hi_r) in zip(
+                            lo.get_xdata(), lo.get_ydata(),
+                            hi.get_xdata(), hi.get_ydata()):
+                    # Rotate caps to be perpendicular to the error bars
+                    rotation = (lo_theta + hi_theta)/2
```

Comment:
I suppose the rotation angle should not be centered between lo and hi, but rather be defined by the theta data value. - Otherwise, e.g. extending the low value would increasingly tilt the errorbar. 

t.b.c. This may already be wrong in the calculation of the bar direction itself, not only in the caps here.